### DISCUSSION TITLE
Why is JSON data for props also included inline in SSG generated html files?

### DISCUSSION THREAD

**Discussion by moop-moop** [vercel/next.js#20108] [Help]
Assuming exclusive use of getStaticProps and getStaticPaths...
1. JSON data for a page is included in separate JSON file for link prefetching.
2. The html file built by the Next.js SSG build includes the content rendered directly in the HTML already.

What is the need for the JSON data for the props to also be included inline?
Thus increasing the html file size etc.
I know passing only truly needed/minimal props would reduce the data size, but why in included inline in the first place? 

Unless we are doing something wrong?
https://patient.uwhealth.org/

**Comment by michrome** [comment]
From [Automatic Static Optimization](https://nextjs.org/docs/advanced-features/automatic-static-optimization):

> **Statically generated pages are still reactive**: Next.js will hydrate your application client-side to give it full interactivity.
